sda-admin/file/file.go

```diff
@@ -0,0 +1,66 @@
+package file
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/neicnordic/sensitive-data-archive/sda-admin/helpers"
+)
+
+type RequestBodyFileIngest struct {
+	Filepath string `json:"filepath"`
+	User     string `json:"user"`
+}
+
+type RequestBodyFileAccession struct {
+	AccessionID string `json:"accession_id"`
+	Filepath    string `json:"filepath"`
+	User        string `json:"user"`
+}
+
+// FileIngest triggers the ingestion of a given file
+func FileIngest(api_uri, token, username, filepath string) error {
+
+	url := api_uri + "/file/ingest"
+
+	requestBody := RequestBodyFileIngest{
+		Filepath: filepath,
+		User:     username,
+	}
+
+	jsonBody, err := json.Marshal(requestBody)
+	if err != nil {
+		return fmt.Errorf("failed to marshal JSON, reason: %v", err)
+	}
+
+	_, err = helpers.PostRequest(url, token, jsonBody)
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
```suggestion
func Ingest(apiURL, token, username, filepath string) error {
	url, err := url.Parse(apiURL)
	if err != nil {
		return err
	}

	url.Path = path.Join(url.Path, "/file/ingest")

	requestBody := RequestBodyFileIngest{
		Filepath: filepath,
		User:     username,
	}

	jsonBody, err := json.Marshal(requestBody)
	if err != nil {
		return fmt.Errorf("failed to marshal JSON, reason: %v", err)
	}

	_, err = helpers.PostRequest(url.String(), token, jsonBody)
	if err != nil {
		return err
	}

	return nil
}
```
This function is part of the `file` package, as such any calls to this function will be prefixed with `file.` so having `File` as the start of the name makes it stutter `file.FileIngest`

When joining URIs and paths use the library helpers to ensure you get a proper string in the end, without bad characters or double slashes.

Always have a blank line before any return unless it is the only statement in an if block.

Reply:
using path.Join() to concatenate url may have problems on Windows. I have modified the function names and also using `url.Parse()` as you suggested in the commit 6952f7d

Reply:
If a windows environment will mess things up make use of the standard libraries to overcome that.
`filepath.Toslash` is very convenient to ensure that the final path is UNIX styled.

You current solution does not fix the issue of getting double slashes in the final path that `path.Join` does.

Reply:
But the proper way of figuring out what needs to be done is to create a test case where you check that the final URL is correctly formatted

Since we will have to set up an GH action to test this app in a windows environment anyway.

Reply:
It makes sense to me. I will create a commit to fix the issues of double slash.
However, if we will be the only user for `sda-admin`, I guess we don't need to spend too much effort for over engineering.

Reply:
`We` are not the main target user of this app, the helpdesk staff and all repository managers in GDI are.

Reply:
I think I was wrong in the previous message. It seem `path.Join` always uses '/' as path separators, even on Windows. I have implemented changes in the commit 34e82e5